File: 14. Longest Common Prefix/Solution.java
// LeetCode #14: Longest Common Prefix

/*
Write a function to find the longest common prefix string amongst an array of strings.
*/

public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0)
            return "";
            
        for (int i = 0; i < strs[0].length(); i++) {
            char letter = strs[0].charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if (i >= strs[j].length() || letter != strs[j].charAt(i))
                    return strs[0].substring(0, i);
            }
        }
        
        return strs[0];
    }
}

File: 14. Longest Common Prefix/Solution.cpp
// LeetCode #14: Longest Common Prefix

/*
Write a function to find the longest common prefix string amongst an array of strings.
*/

class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) {
        if (strs.size() == 0)
            return "";
            
        for (int i = 0; i < strs[0].size(); i++) {
            char letter = strs[0][i];
            for (int j = 1; j < strs.size(); j++) {
                if (i >= strs[j].size() || letter != strs[j][i])
                    return strs[0].substr(0, i);
            }
        }
        return strs[0];
        
    }
};